nomad/state/state_store.go

```diff
@@ -1907,8 +1907,36 @@ func (s *StateStore) deleteJobScalingPolicies(index uint64, job *structs.Job, tx
 }
 
 func (s *StateStore) deleteJobSubmission(job *structs.Job, txn *txn) error {
-	_, err := txn.DeleteAll("job_submission", "by_jobID", job.Namespace, job.ID)
-	return err
+	// find submissions associated with job
+	remove := *set.NewHashSet[*structs.JobSubmission, string](6)
```

Comment:
Is there a const we can use for 6 (because I assume it's meant to match the max number of versions we intend to keep around)

Reply:
Actually can you use `txn.DeletePrefix("job_submission", "id_prefix", job.Namespace, job.ID)` here instead?

Reply:
Oh no. :flushed: Is it ever possible to safely use `DeletePrefix` on indexes with user controlled fields?!